core/src/wallet/cosmos/explorers/GaiaCosmosLikeBlockchainExplorer.cpp

```diff
@@ -213,23 +211,25 @@ void addMsgFeesTo(cosmos::Transaction &transaction, const T &node)
 }
 
 template <typename T>
-void parseTransactionWithPosttreatment(const T &node, cosmos::Transaction &transaction)
+void GaiaCosmosLikeBlockchainExplorer::parseTransactionWithPosttreatment(
+    const T &node, cosmos::Transaction &transaction) const
 {
-    rpcs_parsers::parseTransaction(node, transaction);
+    rpcs_parsers::parseTransaction(node, transaction, _currency);
     addMsgFeesTo(transaction, node);
 }
 
-}  // namespace
 
 FuturePtr<cosmos::TransactionsBulk> GaiaCosmosLikeBlockchainExplorer::getTransactions(
     const CosmosLikeBlockchainExplorer::TransactionFilter &filter, int page, int limit) const
 {
+    // NOTE : the amount of memory involved here asks for a second pass to make
+    // sure that the least amount of copies is done.
```

Comment:
Made my head explode, but well done.
Maybe just add a couple comments along the way to help follow the big steps.